src/main/java/com/o19s/es/ltr/feature/store/PrecompiledExpressionFeature.java

```diff
@@ -65,13 +76,41 @@ public String name() {
 
     @Override
     public Query doToQuery(LtrQueryContext context, FeatureSet set, Map<String, Object> params) {
-        return new DerivedExpressionQuery(set, expression);
+        List<String> missingParams = queryParams.stream()
+                .filter((x) -> params == null || !params.containsKey(x))
+                .collect(Collectors.toList());
+        if (!missingParams.isEmpty()) {
+            String names = missingParams.stream().collect(Collectors.joining(","));
+            throw new IllegalArgumentException("Missing required param(s): [" + names + "]");
+        }
+
+        Map<String, Double> queryParamValues = getQueryParamValues(params);
+        return new DerivedExpressionQuery(set, expression, queryParamValues);
+    }
+
+    /* if a param is an expression variable then we need to use it as a `variable` in lucene expression exposed via Bindings */
```

Comment:
nitpick: this can be valid javadoc comment

Reply:
done